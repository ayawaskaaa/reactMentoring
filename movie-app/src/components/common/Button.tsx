import React from "react";
import PropTypes from "prop-types";

interface ButtonProps {
    onClick(e);

    type: "button" | "submit" | "reset" | undefined;
    text: string;
    classes: string;
}

function Button(props: ButtonProps) {

    return (
        <>
            <button type={props.type} onClick={props.onClick}
                    className={props.classes}>{props.text}</button>
        </>
    )
}

Button.prototype = {
    onClick: PropTypes.func,
    text: PropTypes.string,
    classes: PropTypes.string,
    type: PropTypes.string,
}
export default Button;