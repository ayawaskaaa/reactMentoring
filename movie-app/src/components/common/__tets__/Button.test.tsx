import React from 'react';
import {cleanup, fireEvent, render, screen} from '@testing-library/react';
import Button from "../Button";

describe("test Button component", () => {
    afterEach(cleanup)
    it('should render correctly', function () {
        const mockOnClick = jest.fn();
        const type = 'button'
        const text = 'someText';
        const classes = 'someClass'
        const {asFragment} = render(<Button onClick={mockOnClick} classes={classes} text={text} type={type}/>)
        expect(asFragment).toMatchSnapshot();
        const element = screen.getByText(text);
        fireEvent.click(element)
        expect(element).not.toBeNull();
        expect(element.textContent).toBe(text);
        expect(element.getAttribute("type")).toBe(type);
        expect(element.getAttribute("class")).toBe(classes);
        expect(mockOnClick).toBeCalled();
    });
})