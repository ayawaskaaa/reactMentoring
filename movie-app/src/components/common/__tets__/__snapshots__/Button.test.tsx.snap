// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Button component should render correctly 1`] = `
<div>
  <button
    class="someClass"
    type="button"
  >
    someText
  </button>
</div>
`;
